tests/utils_tests/test_q.py

```diff
@@ -0,0 +1,98 @@
+from unittest import TestCase
```

Comment:
I think a better location could be something like `test/queries/test_q.py`. I know that `Q` inherits from `utils.tree`, but I don't think it's a natural place to look for these tests.